src/ChluIPFS.js

```diff
@@ -167,37 +167,98 @@ class ChluIPFS {
         }
     }
 
-    async readReviewRecord(multihash) {
+    async getLastReviewRecordUpdate(db, multihash) {
+        this.logger.debug('Checking for review updates for ' + multihash);
+        let dbValue = multihash, updatedMultihash = multihash, path = [multihash];
+        while (dbValue) {
+            dbValue = await db.get(dbValue);
+            if (typeof dbValue === 'string') {
+                if (path.indexOf(dbValue) < 0) {
+                    updatedMultihash = dbValue;
+                    path.push(dbValue);
+                    this.logger.debug('Found forward pointer from ' + multihash + ' to ' + updatedMultihash);
+                } else {
+                    throw new Error('Recursive references detected in this OrbitDB: ' + db.address.toString())
+                }
+            }
+        }
+        if (multihash != updatedMultihash) {
+            this.logger.debug(multihash + ' updates to ' + updatedMultihash);
+            return updatedMultihash;
+        } else {
+            this.logger.debug('no updates found for ' + multihash);
+        }
+    }
+
+    async notifyIfReviewIsUpdated(db, multihash, notifyUpdate) {
+        const updatedMultihash = await this.getLastReviewRecordUpdate(db, multihash);
+        if (updatedMultihash) {
+            // TODO: Check that the update is valid first
+            notifyUpdate(multihash, updatedMultihash);
+        }
+    }
+
+    async findLastReviewRecordUpdate(multihash, notifyUpdate) {
+        const reviewRecord = await this.getReviewRecord(multihash);
+        if (reviewRecord.orbitDb) {
+            const db = await this.openDbForReplication(reviewRecord.orbitDb);
+            db.events.once('replicated', () => this.notifyIfReviewIsUpdated(db, multihash, notifyUpdate));
+            this.notifyIfReviewIsUpdated(db, multihash, notifyUpdate);
+        }
+    }
+
+    async getReviewRecord(multihash){
         const dagNode = await this.ipfs.object.get(this.utils.multihashToBuffer(multihash));
-        const buffer = dagNode.Data;
+        const buffer = dagNode.data;
         const messages = protons(require('../src/utils/protobuf'));
         const reviewRecord = messages.ReviewRecord.decode(buffer);
-        // TODO: validate
         return reviewRecord;
     }
 
-    async storeReviewRecord(reviewRecord){
+    async readReviewRecord(multihash, notifyUpdate = null) {
```

Comment:
@fazo96 - why did you choose to use a callback approach here and not promises? That way the caller can chose to wait for it be resolved if they want to receive confirmation of replication

Reply:
when you call this fn, it will return a promise that resolves to the review record that matches to the actual multihash given. The problem is checking for updates: it's very hard to be certain that there are no updates, so I thought about asking the library to check for updates of the review at the given multihash as an optional thing.

If we only resolve the promise when we are sure we have the latest update then the read operation might be too slow and the caller is left hanging there with no data. A callback approach works better because it lets us use the promise to give a faster but maybe not updated response (so that the UI can show something) while checking for updates in the background and eventually updating the UI if something is found.

It's also true that at least with the current way we do things, this node probably already has the review update replicated when you try to read an older version so the callback should be called really fast.

Let me know what you think